File: ts/src/solutions/running-sum-of-1d-array/running-sum-of-1d-array.test.ts
import { InputAndExpectedValue } from '../../input-and-expected-value';
import { runningSum } from './solution';

describe('Running sum of 1d array', () => {
    const inputsAndExpectedValues: InputAndExpectedValue<number[], number[]>[] = [
        {
            i: [],
            e: []
        },
        {
            i: [1, 2, 3, 4],
            e: [1, 3, 6, 10]
        },
        {
            i: [1, 1, 1, 1, 1],
            e: [1, 2, 3, 4, 5]
        },
        {
            i: [3, 1, 2, 10, 1],
            e: [3, 4, 6, 16, 17]
        },
    ];

    it(`should return the running sum`, () => {
        for (const iAndE of inputsAndExpectedValues) {
            expect(runningSum(iAndE.i)).toEqual(iAndE.e);
        }
    });
});

File: go/src/solutions/running-sum-of-1d-array/running-sum-of-1d-array_test.go
package runningSum1D

import (
	inputAndExpectedValue "go-coding-challenges/src"
	"testing"

	"github.com/forestgiant/sliceutil"
)

var inputsAndExpectedValues = []inputAndExpectedValue.InputAndExpectedValue[[]int, []int]{
	{I: []int{}, E: []int{}},
	{I: []int{1, 2, 3, 4}, E: []int{1, 3, 6, 10}},
	{I: []int{1, 1, 1, 1, 1}, E: []int{1, 2, 3, 4, 5}},
	{I: []int{3, 1, 2, 10, 1}, E: []int{3, 4, 6, 16, 17}},
}

func TestRunningSum(t *testing.T) {
	for _, v := range inputsAndExpectedValues {
		input := v.I
		expected := v.E
		result := runningSum(input)
		if !sliceutil.Compare(result, expected) {
			t.Errorf("Expected: %v, Received: %v", expected, result)
		}
	}
}

func BenchmarkFib1Input8(b *testing.B) {
	inputIndex := len(inputsAndExpectedValues) - 1
	for i := 0; i < b.N; i++ {
		runningSum(inputsAndExpectedValues[inputIndex].I)
	}
}
